docs/DeveloperGuide.adoc

```diff
@@ -239,85 +238,147 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+** Only the names of the `Person` is recorded in an `Event`, as the name should be unique and uneditable.
 
-The following sequence diagram shows how the undo operation works:
+Storing only the *name* of `Person` in the `Attendees` object of `Event` saves memory storage and facilitates the `select` feature. As we select a person, we can simply iterate through the event list and check whether the person's name is in the attendees list.
 
-image::UndoRedoSequenceDiagram.png[width="800"]
+We can also have a summarized content of persons (their names) involved in an event. Obtaining the full information of the list of persons in a particular event is not required in the current version of project, but can still be implemented easily with the current `Attendee` structure. This can be done by filtering through the address book based on the person names present in `Attendees`. Thus, there is no need to store any event information in the `Person` object.
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+==== Execution of Command
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+Given below is an example usage scenario and how the invite mechanism works.
+
+Step 1. The user executes `invite 1 to/1` to invite the 1st person in the address book to the 1st event in the event list. The command text is passed to an instance of the `LogicManager` class.
+
+Step 2. The `LogicManager` instance calls `AddressBookParser#parsecommand`, which parses the `invite` command phrase.
+
+Step 3. `InviteCommandParser#parse` parses the person and event index. An instance of the `InviteCommand` encapsulating the two indices information is then returned.
+
+Step 4. `Logic Manager` then executes this `InviteCommand` by calling `InviteCommand#execute`.
+
+Step 5. The filtered person list and event list is first obtained by calling `PersonModel#getFilteredPersonList` and `EventModel#getFilteredEventList`. Based on the indices, the `Person` and `Event` is selected.
+
+Step 6. Next, the `Person` object's name is obtained and added to the obtained `Attendees` object. The new `Attendees` object is then added to a new copy of the `Event` object.
+
+Step 7. The new `Event` object is updated to the model by calling `Model#updateEvent`.
+
+Step 8. The `InviteCommand#execute` finally returns a `CommandResult` with a success message.
+
+The sequence diagram below illustrates the execution of the `invite` command.
+
+image::InviteSequenceDiagram.png[width="800"]
```

Comment:
Good job here on the sequence diagram. 


- drew proper lifelines for each object
- specified invocation method
- use dashed arrows for return calls.  
- clear separation of Logic classes and Model classes


However, note that: 
To reduce clutter, **activation bars** and **return arrows** **may be omitted** if they do not result in ambiguities or loss of information. Informal operation descriptions such as those given in the example below can be used, if more precise details are not required for the task at hand.